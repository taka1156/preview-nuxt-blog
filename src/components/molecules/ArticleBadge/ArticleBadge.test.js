import { shallowMount } from '@vue/test-utils';
import ArticleBadge from './ArticleBadge.vue';
import { dummyCategoryBadge, dummyTagBadge } from '@/__testdata__/testdata.js';

describe('ArticleBadge', () => {
  const articleBadge = propsData =>
    shallowMount(ArticleBadge, {
      propsData: {
        ...propsData
      }
    });

  it('値がpropsに反映されているか: CategoryBadge', () => {
    const wrapper = articleBadge(dummyCategoryBadge);
    expect(wrapper.vm.badge).toBe(dummyCategoryBadge.badge);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('値がpropsに反映されているか: TagBadge', () => {
    const wrapper = articleBadge(dummyTagBadge);
    expect(wrapper.vm.badge).toBe(dummyTagBadge.badge);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
