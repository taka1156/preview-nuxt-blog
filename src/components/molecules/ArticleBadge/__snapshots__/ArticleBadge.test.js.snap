// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleBadge 値がpropsに反映されているか: CategoryBadge 1`] = `
"<div>
  <div class=\\"article-badge article-badge--extend\\">
    <base-text-stub>
      ダミーカテゴリー
    </base-text-stub>
    <base-img-stub imgurl=\\"http://placehold.jp/150x150.png\\" imgalt=\\"ダミーカテゴリーの画像\\" size=\\"sm\\"></base-img-stub>
  </div>
</div>"
`;

exports[`ArticleBadge 値がpropsに反映されているか: TagBadge 1`] = `
"<div>
  <div class=\\"article-badge article-badge--extend\\">
    <base-text-stub>
      ダミータグ
    </base-text-stub>
    <base-img-stub imgurl=\\"http://placehold.jp/150x150.png\\" imgalt=\\"ダミータグの画像\\" size=\\"sm\\"></base-img-stub>
  </div>
</div>"
`;
